cmd/system-probe/modules/gpu_monitoring.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+package modules
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/utils"
+	"github.com/DataDog/datadog-agent/pkg/gpu"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var _ module.Module = &GPUMonitoringModule{}
+var gpuMonitoringConfigNamespaces = []string{gpu.GPUConfigNS}
+
+// GPUMonitoring Factory
+var GPUMonitoring = module.Factory{
+	Name:             config.GPUMonitoringModule,
+	ConfigNamespaces: gpuMonitoringConfigNamespaces,
+	Fn: func(_ *sysconfigtypes.Config, _ module.FactoryDependencies) (module.Module, error) {
+		t, err := gpu.NewProbe(gpu.NewConfig(), nil)
```

Comment:
`gpu.NewConfig()` 

```go
func NewConfig() *Config {
	cfg := pkgconfigsetup.SystemProbe()
	sysconfig.Adjust(cfg)

	return &Config{
		Config:                          ebpf.NewConfig(),
		ScanTerminatedProcessesInterval: 5 * time.Second,
	}
}
```

Seems to get the systemprobe config, could we pass the `sysconfigtypes.Config` from the function arguments from the `Fn` field?